modules/library/main/src/test/java/org/geotools/filter/function/QuantileFunctionTest.java

```diff
@@ -235,4 +236,19 @@ public void testConstantValuesString() {
         assertEquals(1, classifier.getValues(0).size());
         assertEquals("abc", classifier.getValues(0).iterator().next());
     }
+
+    @Test
+    public void testEvaluateWithPercentages() throws Exception {
```

Comment:
Please add a test for unequally distributed classes, due to the data itself.

Reply:
tests added